src/console_node/logRotation.go

```diff
@@ -167,9 +193,13 @@ func updateLogRotateConf() {
 		# }
 	*/
 
+	// put a lock on the current nodes while writing the file
```

Comment:
I don't think this 'locks' them, you just get a current copy that isn't going to change under you as you iterate over them.